components/user_settings/display/manage_languages/manage_languages.jsx

```diff
@@ -21,15 +24,49 @@ export default class ManageLanguage extends React.Component {
 
     constructor(props) {
         super(props);
+        const locales = I18n.getLanguages();
+        const userLocale = props.locale;
+        const selectedOption = {value: locales[userLocale].value, label: locales[userLocale].name};
+        this.reactSelectContainer = React.createRef();
 
         this.state = {
             locale: props.locale,
+            selectedOption,
             isSaving: false,
+            openMenu: false,
         };
     }
 
-    setLanguage = (e) => {
-        this.setState({locale: e.target.value});
+    componentDidMount() {
+        if (this.reactSelectContainer.current) {
+            this.reactSelectContainer.current.addEventListener('keydown', this.handleContainerKeyDown);
```

Comment:
Why not use the `onKeyDown` prop for this instead of managing the event listener manually?

Reply:
Because that does not work as expected, it does not pass the `event` that we expect. There is another way to detect that, which is `https://github.com/JedWatson/react-select/issues/944`. 

As you can see, the `onKeyDown` that we want is not to be run on the `react-select`.

Reply:
Oh, I see. That's annoying. Hopefully it'll get sorted out at some point.